snapcraft/plugins/catkin_tools.py

```diff
@@ -0,0 +1,124 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015-2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The catkin_tools build plugin is useful for building ROS parts.
+
+This plugin relies on the catkin plugin as the catkin plugin performs
+configuration that allows catkin_tools to run.  This plugin uses the
+same keywords and configurations as the catkin plugin, the difference
+is the installation of and using catkin_tools to build.
+"""
+
+import os
+import logging
+
+import snapcraft.plugins.catkin
+from snapcraft.plugins.catkin import (
+    Compilers,
+)
+
+import snapcraft
+
+logger = logging.getLogger(__name__)
+
+
+class CatkinToolsPlugin(snapcraft.plugins.catkin.CatkinPlugin):
+    def __init__(self, name, options, project):
+        super().__init__(name, options, project)
+        self.build_packages.extend(['python-catkin-tools'])
+
+    def _prepare_build(self):
```

Comment:
I feel that this method is too long. Maybe it can be split into `_clean_catkin`, `_add_catkin_profile` and `_install_catkin_packages`